akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/ParameterDirectivesSpec.scala

```diff
@@ -50,6 +50,18 @@ class ParameterDirectivesSpec extends FreeSpec with GenericRoutingSpec with Insi
         }
       }
     }
+    "supply chaining of unmarshallers" in {
+      case class UserId(id: Int)
+      case class AnotherUserId(id: Int)
+      val UserIdUnmarshaller = Unmarshaller.strict[Int, UserId](UserId)
+      val AnotherUserIdUnmarshaller = Unmarshaller.strict[UserId, AnotherUserId](userId ⇒ AnotherUserId(userId.id))
+      Get("/?id=45") ~> {
+        parameter('id.as[Int].as(UserIdUnmarshaller)) { echoComplete }
```

Comment:
Hmm... yeah we've indeed fixed the specific case.

I was wondering if we'd be able to pull of being able to chain like this:

```scala
implicit val UserIdUnmarshaller = Unmarshaller.strict[Int, UserId](UserId)
implicit val AnotherUserIdUnmarshaller = Unmarshaller.strict[UserId, AnotherUserId](userId ⇒ AnotherUserId(userId.id))

parameter('id.as[Int].as[UserId]) { echoComplete }
```

but this gets much more tricky... if you'd like to give it a shot that'd be awesome, let me know if that's something you'd like to try out to make possible. I think we'll need special casing such that the FSU is only pulled in if we're indeed from String, and otherwise we need to keep around more things hm